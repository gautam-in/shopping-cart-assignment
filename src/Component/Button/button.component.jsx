import React from "react";
import "./button.styles.scss";

function Button({ text, onClick = null }) {
  return (
    <div className="button-cont">
      <button className="button-prm" onClick={() => onClick && onClick()} aria-label={text}>
        {text}
      </button>
    </div>
  );
}

export default Button;
