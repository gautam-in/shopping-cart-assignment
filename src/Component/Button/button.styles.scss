.button-cont {
  display: flex;
  justify-content: center;
  .button-prm {
    background-color: rgb(211, 62, 62);
    padding: 1vh;
    font-size: medium;
    min-width: 10vw;
    color: #fff;
    border: none;
    border-radius: 5px;
  }
  .button-prm:active {
    background-color: rgb(231, 81, 81);
  }
}

@media (max-width: 425px) {
  .button-cont {
    .button-prm {
      font-size: 1vh;
    }
  }
}
