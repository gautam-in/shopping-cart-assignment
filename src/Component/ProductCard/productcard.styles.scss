.card-container {
  margin-top: 5vh;
  width: 15vw;
  padding: 2vw;
  box-shadow: -2px 4px 10px -6px rgb(209, 202, 202);
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .thumbnail-container {
    display: flex;
    justify-content: center;
    .product-img {
      width: 10vw;
    }
  }

  .card-desc {
    background-color: rgb(235, 235, 235);
    padding: 1vw;
    font-size: small;
    margin-top: 1vh;
    margin-bottom: 1vh;
  }

  .footer-container {
    display: flex;
    justify-content: space-between;
    align-items: center;

    .price-tag::before {
      content: "MRP Rs.";
      font-size: small;
    }
  }
}

@media (max-width: 1024px) {
  .card-container {
    display: flex;
    flex-direction: wrap;
    width: 20vw;
  }
}

@media (max-width: 768px) {
  .card-container {
    display: flex;
    flex-direction: wrap;
    width: 30vw;
  }
}

@media (max-width: 425px) {
  .card-container {
    display: flex;
    flex-direction: wrap;
    width: 70vw;
    margin-top: 2vh;
    font-size: small;
  }
}
