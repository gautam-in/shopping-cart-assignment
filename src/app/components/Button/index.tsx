/**
 *
 * Button
 *
 */
import React, { memo } from 'react';
import { Button as MuiButton, ButtonProps } from '@mui/material';
import { createTheme, ThemeProvider, styled } from '@mui/material/styles';

interface Props extends ButtonProps {
  label: string;
}

const theme = createTheme({
  palette: {
    primary: {
      main: '#F92A53',
    },
  },
});

export const Button = memo((props: Props) => {
  const { label, ...rest } = props;

  return (
    <ThemeProvider theme={theme}>
      <CustomButton variant="contained" {...rest}>
        {label}
      </CustomButton>
    </ThemeProvider>
  );
});

const CustomButton = styled(MuiButton)`
  width: 60%;
  border-radius: 0px;
  text-transform: capitalize;
`;
