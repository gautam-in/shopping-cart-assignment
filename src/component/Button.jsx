import "../styles/button.scss";

const BUTTON_TYPE_CLASSES = {
  max: "max",
  buy: "buy",
  checkout: "checkout",
};
const Button = ({ children, buttonType, ...otherProps }) => {
  return (
    <button
      className={`button-container ${BUTTON_TYPE_CLASSES[buttonType]}`}
      {...otherProps}
    >
      {children}
    </button>
  );
};

export default Button;
