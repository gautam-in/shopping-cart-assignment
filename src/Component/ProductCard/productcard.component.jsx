import React from "react";
import "./productcard.styles.scss";
import Button from "../Button/button.component";

function ProductCard({ name, imgURL, desc, price, additem }) {
  const onClickBuy = () => {
    additem();
  };
  return (
    <div className="card-container">
      <h3>{name}</h3>
      <div className="thumbnail-container">
        <img
          className="product-img"
          src={`${process.env.PUBLIC_URL}${imgURL}`}
          alt={name}
        />
      </div>
      <div className="card-desc"> {desc} </div>
      <div className="footer-container">
        <div className="price-tag"> {price} </div>
        <Button text="Buy" onClick={onClickBuy} />
      </div>
    </div>
  );
}

export default ProductCard;
