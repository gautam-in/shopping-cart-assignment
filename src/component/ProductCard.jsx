import axios from "axios";
import { useDispatch, useSelector } from "react-redux";
import { selectCartItems } from "../store/slices/cart/cart.selector";
import { addItemsToCart } from "../store/slices/cart/cart.action";
import Button from "./Button";
import "../styles/product-card.scss";

const ProductCard = ({ product }) => {
  const { id, name, imageURL, description, price } = product;
  const dispatch = useDispatch();
  const cartItems = useSelector(selectCartItems);
  const handleBuy = async () => {
    try {
      await axios.post("http://localhost:8000/addToCart").then((res) => {
        if (res.data.response === "Success") {
          dispatch(addItemsToCart(cartItems, product));
        }
      });
    } catch (error) {
      alert("Something went wrong");
      console.log("There is some error while buying this item", error.message);
    }
  };

  return (
    <div className="product-card">
      <div className="product-card__details">
        <h1 className="product-card-title">{name}</h1>
        <div className="product-card__desc-container">
          <div className="product-card__img-container">
            <img
              className="product-card-img"
              src={require(`../../src${imageURL}`)}
              alt={`${name}`}
            />
          </div>
          <p className="product-card-desc">{description}</p>
        </div>
      </div>
      <div className="product-card__bottom">
        <span className="product-card-price">price Rs{price}</span>
        <Button
          onClick={handleBuy}
          buttonType="max"
          type="button"
          aria-label={`Buy ${name}`}
        >
          Buy Now
          <span
            className="product-card-price--mobile"
            style={{ color: "white" }}
          >
            @ MSRP Rs {price}
          </span>
        </Button>
      </div>
    </div>
  );
};

export default ProductCard;

// <div className="product-card" key={id}>
//   <div className="product-details">
//     <h2 className="product-name">{name}</h2>
//     <img
//       src={require(`../../src${imageURL}`)}
//       alt={`${name}`}
//       className="product-image"
//     />
//     <span className="product-description">{description}</span>
//   </div>
//   <div className="product-action">
//     <span className="price">MRP Rs. {price}</span>
//     <Button buttonType="buy" onClick={handleBuy}>
//       But now
//     </Button>
//   </div>
// </div>
