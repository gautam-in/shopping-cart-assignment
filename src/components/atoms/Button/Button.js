import React from "react";
import "./Button.scss";

export default function Button({
  children,
  className = "",
  onClick,
  ...props
}) {
  return (
    <button
      className={`${className} button-wrapper`}
      onClick={onClick}
      {...props}
    >
      {children}
    </button>
  );
}
