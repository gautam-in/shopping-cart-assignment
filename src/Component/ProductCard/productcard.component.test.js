import { shallow } from "enzyme";
import React from "react";
import Button from "../Button/button.component";
import ProductCard from "./productcard.component";

describe("expects to test major use cases of product card", () => {
  const wrapper = shallow(
    <ProductCard
      name={"Apple"}
      imgURL={"/static/apple.png"}
      desc={"Red fruit"}
      price={180}
    />
  );
  it("expects to  return the component with all the props", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("expects to return the component with all the given props", () => {
    expect(wrapper.find("h3").prop("children")).toBe("Apple");
    expect(wrapper.find("img").prop("src")).toBe(
      `${process.env.PUBLIC_URL}/static/apple.png`
    );
    expect(wrapper.containsMatchingElement(<Button />)).toBe(true);
  });
});
