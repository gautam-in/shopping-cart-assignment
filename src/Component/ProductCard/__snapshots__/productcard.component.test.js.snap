// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to test major use cases of product card expects to  return the component with all the props 1`] = `ShallowWrapper {}`;
